### DISCUSSION TITLE
Calendar should differentiate between reaching or missing the target

### DISCUSSION THREAD

**Discussion by zareami10** [iSoron/uhabits#918] [Feature Requests]
_This is somewhere in between a bug/feature request so I opted to discuss it here at the end._

So as the title says, calendar's state is currently binary only, which doesn't make a lot of sense when it comes to the new Measurable tasks. I think it would be much more helpful to know whether you have hit your target or not.

Another approach might be using gradients instead of only 3 states (e.g. more saturated color as the amount gets closer to the target, like the Frequency chart), but I'm not sure how useful this might be.

**Comment by Silviopina** [comment]
What about 1/2 filled square to indicate "did some" but not hit "full target"?

**Comment by lily-peng** [comment]
I agree, like a darker black/grey for either days with no data or days where the goal was not met.

And it should apply to the widget too.
